### ISSUE TITLE
`dotnet publish` is not idempotent for single file publishing when project contains App.config

### ISSUE THREAD

**Issue by Blackclaws** [dotnet/sdk#38128]
### Describe the bug
When running `dotnet publish -o out` for SingleFile publishing running it a second time changes the output when the project contains an `App.config` file.

The first output erroneously contains a `AppName.dll.config` file.

This happens on Windows an Linux platforms, I don't have a mac to test on but I assume its similar.

### To Reproduce
1. Clone repo [ReproducerRepo](https://github.com/Blackclaws/PublishDeletionReproducer)
2. Run `dotnet publish -o out` 
3. Check contents of `out` folder
4. Rerun `dotnet publish -o out`
5. Check contents again and notice change

Deleting the `out` folder restarts the process.

### Further technical details
```
.NET SDK:
 Version:           8.0.101
 Commit:            6eceda187b
 Workload version:  8.0.100-manifests.69afb982

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  23.10
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /usr/share/dotnet/sdk/8.0.101/

.NET workloads installed:
 Workload version: 8.0.100-manifests.69afb982
There are no installed workloads to display.

Host:
  Version:      8.0.1
  Architecture: x64
  Commit:       bf5e279d92

.NET SDKs installed:
  6.0.418 [/usr/share/dotnet/sdk]
  8.0.101 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.26 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.15 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.26 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.15 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/usr/share/dotnet]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
  ```

**Comment by Blackclaws** [comment]
This seems to apply also to any content added via:
```
<Content Include="Images/*">
    <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
</Content>
```

Its there on the first publish but deleted on subsequent ones.

**Comment by EdLichtman** [comment]
Hey! I was just searching this issue because I'm finding that I'm running into it again, and happened upon this untriaged issue, so I thought I'd just pop in and help:

https://github.com/dotnet/sdk/issues/37222

I don't know if this is exactly the same thing you're seeing, but basically, when I dotnet publish, I have a <None Include... CopyToPublishDirectory="always"> item, and it seems like when I run dotnet publish the first time, it gets copied, but when I run dotnet publish the second time, all I see in the PublishDir is the .exe file.

So basically, I'm just adding:
```
<ItemGroup>
  <Content Update="*">
    <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
  </Content>
```
To the bottom, rather than adding that to every single ItemGroup.

**Comment by ltjax** [comment]
Thank you, I had this problem as well, even without an `app.config` file. According to the binlog, the target `_IncrementalCleanPublishDirectory` deletes my .csv / .ini / native .dll files on the second `dotnet publish` run.
I can confirm that `ExcludeFromSingleFile` works as a work-around for my .csv / .ini resource files. However, native .dll files of dependencies were still deleted by `_IncrementalCleanPublishDirectory`. For that, IncludeNativeLibrariesForSelfExtract is a workaround.
